### ISSUE TITLE
I2C: Drivers must have a  implementation resetting the peripheral to idle state.

### ISSUE THREAD

**Issue by bugadani** [esp-rs/esp-hal#2774]
Is it an exception? For now, I can't see any reason why not to do a `Drop` implementation for this. IDF also has some code for this.

**Comment by MabezDev** [comment]
I don't think this is an exception, we should do it. We can technically do this post 1.0, but I think it would be a good idea to have it done beforehand.

**Comment by bugadani** [comment]
Franky I have no idea what was in past Daniel's head. :(

**Comment by MabezDev** [comment]
I don't think there is anything actionable here actually, reset on drop is managed by the internal guard.
